public/app/core/components/RolePicker/RolePicker.tsx

```diff
@@ -82,14 +82,13 @@ export const RolePicker = ({
      * both (the role picker menu and its sub menu) aligned to the left edge of the input.
      * Otherwise, it aligns the role picker menu to the right.
      */
-    if (
-      window.innerWidth - right < currentRolePickerWidth &&
-      currentRolePickerWidth < 2 * ROLE_PICKER_SUBMENU_MIN_WIDTH
-    ) {
-      horizontal = offsetHorizontal;
+    if (left + ROLE_PICKER_MAX_MENU_WIDTH > window.innerWidth) {
+      horizontal = window.innerWidth - right;
```

Comment:
I push menu to the right side of input to get a bit more space at the left (it's also easier to calculate).